components/public-api-server/pkg/proxy/prometheusmetrics_test.go

```diff
@@ -0,0 +1,38 @@
+package proxy
+
+import (
+	"context"
+	"github.com/gitpod-io/gitpod/common-go/baseserver"
+	v1 "github.com/gitpod-io/gitpod/public-api/v1"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+	"testing"
+)
+
+func TestConnectionCreationWasTracked(t *testing.T) {
+	// Set up server
+	srv := baseserver.NewForTests(t)
+	baseserver.StartServerForTests(t, srv)
+
+	// Set up Prometheus registry
+	registry := prometheus.NewRegistry()
+	registry.MustRegister(proxyConnectionCreateDurationSeconds)
+
+	// Set up Workspace client
+	ctx := metadata.AppendToOutgoingContext(context.Background(), "authorization", "some-token")
+	conn, err := grpc.Dial(srv.GRPCAddress(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	client := v1.NewWorkspacesServiceClient(conn)
+
+	// Call GetWorkspace
+	client.GetWorkspace(ctx, &v1.GetWorkspaceRequest{
+		WorkspaceId: "some-ID",
+	})
+
+	count, err := testutil.GatherAndCount(registry)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, count)
```

Comment:
@easyCZ So this passes, but not sure if this is the right approach?

Reply:
Nice! Yeah this is good, well done!